test/hotspot/jtreg/compiler/c1/CanonicalizeGetModifiers.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2021, Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test
+ * @author Yi Yang
+ * @summary Canonicalizes Foo.class.getModifiers() with interpreter mode
+ * @library /test/lib
+ * @run main/othervm -Xint
+ *                   -XX:CompileCommand=compileonly,*CanonicalizeGetModifiers.test
+ *                   compiler.c1.CanonicalizeGetModifiers
+ */
+
+/*
+ * @test
+ * @author Yi Yang
+ * @summary Canonicalizes Foo.class.getModifiers() with C1 mode
+ * @requires vm.compiler1.enabled
+ * @library /test/lib
+ * @run main/othervm -XX:TieredStopAtLevel=1 -XX:+TieredCompilation
+ *                   -XX:CompileCommand=compileonly,*CanonicalizeGetModifiers.test
+ *                   compiler.c1.CanonicalizeGetModifiers
+ */
+
+/*
+ * @test
+ * @author Yi Yang
+ * @summary Canonicalizes Foo.class.getModifiers() with C2 mode
+ * @requires vm.compiler2.enabled
+ * @library /test/lib
+ * @run main/othervm -XX:TieredStopAtLevel=4 -XX:-TieredCompilation
```

Comment:
`-XX:TieredStopAtLevel=4` does not work when Tiered is off. You don'ty need this flag.

Reply:
Okay, I learned this from [Valhalla project](https://github.com/openjdk/valhalla/blob/c8d7c8260921ccb2b2e72dcc6cc04144ff7649fc/test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestC1.java#L61). Maybe it's redundant if -XX:-TieredCompilation is disabled.